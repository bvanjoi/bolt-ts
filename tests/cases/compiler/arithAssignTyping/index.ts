// From `github.com/microsoft/TypeScript/blob/v5.7.2/tests/cases/compiler/arithAssignTyping.ts`, Apache-2.0 License

class f { }

f += '';   //~ ERROR: Cannot assign to 'f' because it is a class.
f += 1;    //~ ERROR: Cannot assign to 'f' because it is a class. 
f -= 1;    //~ ERROR: Cannot assign to 'f' because it is a class. 
f *= 1;    //~ ERROR: Cannot assign to 'f' because it is a class. 
f /= 1;    //~ ERROR: Cannot assign to 'f' because it is a class. 
f %= 1;    //~ ERROR: Cannot assign to 'f' because it is a class. 
f &= 1;    //~ ERROR: Cannot assign to 'f' because it is a class. 
f |= 1;    //~ ERROR: Cannot assign to 'f' because it is a class. 
f <<= 1;   //~ ERROR: Cannot assign to 'f' because it is a class.  
f >>= 1;   //~ ERROR: Cannot assign to 'f' because it is a class.  
f >>>= 1;  //~ ERROR: Cannot assign to 'f' because it is a class. 
f ^= 1;    //~ ERROR: Cannot assign to 'f' because it is a class. 