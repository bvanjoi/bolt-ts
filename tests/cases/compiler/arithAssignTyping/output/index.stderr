  × Cannot assign to 'f' because it is a class.
   ╭─[../../tests/cases/compiler/arithAssignTyping/index.ts:5:1]
 5 │ f
   · ─
   ╰────

  × Cannot assign to 'f' because it is a class.
   ╭─[../../tests/cases/compiler/arithAssignTyping/index.ts:6:1]
 6 │ f
   · ─
   ╰────

  × Cannot assign to 'f' because it is a class.
   ╭─[../../tests/cases/compiler/arithAssignTyping/index.ts:7:1]
 7 │ f
   · ─
   ╰────

  × Cannot assign to 'f' because it is a class.
   ╭─[../../tests/cases/compiler/arithAssignTyping/index.ts:8:1]
 8 │ f
   · ─
   ╰────

  × Cannot assign to 'f' because it is a class.
   ╭─[../../tests/cases/compiler/arithAssignTyping/index.ts:9:1]
 9 │ f
   · ─
   ╰────

  × Cannot assign to 'f' because it is a class.
    ╭─[../../tests/cases/compiler/arithAssignTyping/index.ts:10:1]
 10 │ f
    · ─
    ╰────

  × Cannot assign to 'f' because it is a class.
    ╭─[../../tests/cases/compiler/arithAssignTyping/index.ts:11:1]
 11 │ f
    · ─
    ╰────

  × Cannot assign to 'f' because it is a class.
    ╭─[../../tests/cases/compiler/arithAssignTyping/index.ts:12:1]
 12 │ f
    · ─
    ╰────

  × Cannot assign to 'f' because it is a class.
    ╭─[../../tests/cases/compiler/arithAssignTyping/index.ts:13:1]
 13 │ f
    · ─
    ╰────

  × Cannot assign to 'f' because it is a class.
    ╭─[../../tests/cases/compiler/arithAssignTyping/index.ts:14:1]
 14 │ f
    · ─
    ╰────

  × Cannot assign to 'f' because it is a class.
    ╭─[../../tests/cases/compiler/arithAssignTyping/index.ts:15:1]
 15 │ f
    · ─
    ╰────

  × Cannot assign to 'f' because it is a class.
    ╭─[../../tests/cases/compiler/arithAssignTyping/index.ts:16:1]
 16 │ f
    · ─
    ╰────
